File: backend/routes/mentor.py
"""
mentor.py — Mentor Dashboard REST API Handlers (SmartAttend Module 2)
Provides endpoints for mentor profile, assigned mentee list, individual mentee
academic details, attendance analytics, and at-risk alerts.
Access is SCOPED — a mentor can ONLY view data for their assigned students.
"""

from flask import Blueprint, request, jsonify
from backend.models.database import db
from backend.models.user import User
from backend.models.course import Course, Enrollment
from backend.models.attendance import AttendanceRecord
from backend.models.mentor import MentorAssignment

mentor_bp = Blueprint('mentor', __name__, url_prefix='/api/mentor')


# ────────────────────────────────────────────────────────────────
# Helper — resolve mentor_id from query params
# ────────────────────────────────────────────────────────────────
def _resolve_mentor_id():
    return request.args.get('mentor_id', 'F001')


def _get_mentee_ids(mentor_id):
    """Returns list of student_ids assigned to this mentor."""
    assignments = MentorAssignment.query.filter_by(mentor_id=mentor_id).all()
    return [a.student_id for a in assignments]


# ════════════════════════════════════════════════════════════════
#  1. MENTOR PROFILE
# ════════════════════════════════════════════════════════════════
@mentor_bp.route('/profile', methods=['GET'])
def get_mentor_profile():
    mentor_id = _resolve_mentor_id()
    mentor = User.query.filter_by(user_id=mentor_id, role='faculty').first()

    if not mentor:
        mentor = User.query.filter_by(role='faculty').first()

    mentee_count = MentorAssignment.query.filter_by(mentor_id=mentor.user_id if mentor else 'F001').count()

    name = mentor.name if mentor else 'Dr. Priya Sharma'
    name_parts = name.replace('Dr. ', '').replace('Prof. ', '').split()
    initials = ''.join([p[0] for p in name_parts[:2]]).upper()

    return jsonify({
        'mentor_id': mentor.user_id if mentor else 'F001',
        'name': name,
        'department': mentor.department if mentor else 'Computer Science & Engineering',
        'email': mentor.email if mentor else 'p.sharma@smartattend.edu',
        'initials': initials,
        'total_mentees': mentee_count,
        'profile_picture': mentor.profile_picture if mentor else None,
    }), 200


# ════════════════════════════════════════════════════════════════
#  2. MENTEE LIST — Only assigned students
# ════════════════════════════════════════════════════════════════
@mentor_bp.route('/mentees', methods=['GET'])
def get_mentees():
    """Returns list of all students assigned to this mentor with attendance summary."""
    mentor_id = _resolve_mentor_id()
    search = request.args.get('search', '').strip()
    mentee_ids = _get_mentee_ids(mentor_id)

    if not mentee_ids:
        return jsonify({'total': 0, 'mentees': []}), 200

    students_query = User.query.filter(User.user_id.in_(mentee_ids), User.role == 'student')
    if search:
        students_query = students_query.filter(
            (User.name.ilike(f'%{search}%')) |
            (User.enrollment_no.ilike(f'%{search}%'))
        )

    students = students_query.order_by(User.name).all()

    result = []
    for s in students:
        # Overall attendance across all courses
        total_recs = AttendanceRecord.query.filter_by(student_id=s.user_id).count()
        present = AttendanceRecord.query.filter_by(student_id=s.user_id, status='Present').count()
        absent = AttendanceRecord.query.filter_by(student_id=s.user_id, status='Absent').count()
        pct = round((present / total_recs) * 100, 1) if total_recs > 0 else 0.0

        # Count enrolled courses
        enrolled_courses = Enrollment.query.filter_by(student_id=s.user_id).count()

        # Get mentor notes
        assignment = MentorAssignment.query.filter_by(
            mentor_id=mentor_id, student_id=s.user_id
        ).first()

        name_parts = s.name.split()[:2] if s.name else ['?']
        initials = ''.join([n[0] for n in name_parts]).upper()

        result.append({
            'student_id': s.user_id,
            'name': s.name,
            'enrollment_no': s.enrollment_no or s.user_id,
            'email': s.email,
            'program': s.program,
            'semester': s.semester,
            'section': s.section,
            'initials': initials,
            'total_records': total_recs,
            'present': present,
            'absent': absent,
            'attendance_pct': pct,
            'enrolled_courses': enrolled_courses,
            'status': 'Satisfactory' if pct >= 75 else 'At Risk',
            'notes': assignment.notes if assignment else None,
            'profile_picture': s.profile_picture,
        })

    return jsonify({
        'total': len(result),
        'mentees': result,
    }), 200


# ════════════════════════════════════════════════════════════════
#  3. INDIVIDUAL MENTEE DETAIL
# ════════════════════════════════════════════════════════════════
@mentor_bp.route('/mentee/<student_id>', methods=['GET'])
def get_mentee_detail(student_id):
    """Returns full academic detail for a specific mentee. Scoped to mentor."""
    mentor_id = _resolve_mentor_id()
    mentee_ids = _get_mentee_ids(mentor_id)

    # Enforce scope — mentor can only view their assigned students
    if student_id not in mentee_ids:
        return jsonify({
            'error': 'Access denied. This student is not assigned to you.',
        }), 403

    student = User.query.filter_by(user_id=student_id, role='student').first()
    if not student:
        return jsonify({'error': 'Student not found'}), 404

    clean_name = student.name or 'Student'
    name_parts = clean_name.split()[:2] if clean_name else ['?']
    initials = ''.join([n[0] for n in name_parts]).upper()
    enrollment = student.enrollment_no or student.user_id.replace('S-', '')

    profile = {
        'student_id': student.user_id,
        'name': clean_name,
        'enrollment_no': enrollment,
        'email': student.email,
        'program': student.program or 'B.Tech CSE',
        'semester': student.semester or 7,
        'section': student.section or 'B',
        'department': student.department,
        'initials': initials,
        'profile_picture': student.profile_picture,
    }

    # Get assignment info
    assignment = MentorAssignment.query.filter_by(
        mentor_id=mentor_id, student_id=student_id
    ).first()

    return jsonify({
        'profile': profile,
        'assignment': assignment.to_dict() if assignment else None,
    }), 200


# ════════════════════════════════════════════════════════════════
#  4. MENTEE ATTENDANCE — Subject-wise breakdown
# ════════════════════════════════════════════════════════════════
@mentor_bp.route('/mentee/<student_id>/attendance', methods=['GET'])
def get_mentee_attendance(student_id):
    """Returns subject-wise attendance for a mentee. Scoped to mentor."""
    mentor_id = _resolve_mentor_id()
    mentee_ids = _get_mentee_ids(mentor_id)

    if student_id not in mentee_ids:
        return jsonify({'error': 'Access denied.'}), 403

    courses = Course.query.all()
    subjects = []
    total_all = 0
    present_all = 0

    for c in courses:
        # Check if student is enrolled
        enrolled = Enrollment.query.filter_by(student_id=student_id, course_id=c.course_id).first()
        if not enrolled:
            continue

        recs = AttendanceRecord.query.filter_by(student_id=student_id, course_id=c.course_id).all()
        total = len(recs) or c.total_classes or 30
        present = len([r for r in recs if r.status == 'Present'])
        absent = len([r for r in recs if r.status == 'Absent'])
        flagged = len([r for r in recs if r.status == 'Flagged'])
        pct = round((present / total) * 100, 1) if total > 0 else 0.0

        total_all += total
        present_all += present

        subjects.append({
            'course_id': c.course_id,
            'code': c.course_code,
            'name': c.course_name,
            'faculty': c.faculty_name,
            'total': total,
            'present': present,
            'absent': absent,
            'flagged': flagged,
            'pct': pct,
            'status': 'Satisfactory' if pct >= 75 else 'Needs Attention',
        })

    overall_pct = round((present_all / total_all) * 100, 1) if total_all > 0 else 0.0

    return jsonify({
        'student_id': student_id,
        'overall_pct': overall_pct,
        'total_classes': total_all,
        'total_present': present_all,
        'subjects': subjects,
    }), 200


# ════════════════════════════════════════════════════════════════
#  5. MENTEE HISTORY — Attendance history
# ════════════════════════════════════════════════════════════════
@mentor_bp.route('/mentee/<student_id>/history', methods=['GET'])
def get_mentee_history(student_id):
    """Returns recent attendance history for a mentee. Scoped to mentor."""
    mentor_id = _resolve_mentor_id()
    mentee_ids = _get_mentee_ids(mentor_id)

    if student_id not in mentee_ids:
        return jsonify({'error': 'Access denied.'}), 403

    limit = int(request.args.get('limit', 30))
    records = AttendanceRecord.query.filter_by(student_id=student_id)\
                                     .order_by(AttendanceRecord.timestamp.desc())\
                                     .limit(limit).all()

    history = []
    for r in records:
        course = Course.query.filter_by(course_id=r.course_id).first()
        history.append({
            'date': r.timestamp.strftime('%Y-%m-%d') if r.timestamp else '—',
            'subject': course.course_name if course else 'Unknown',
            'code': course.course_code if course else '—',
            'status': r.status,
            'time': r.timestamp.strftime('%I:%M %p') if r.timestamp else '—',
            'confidence': r.confidence_score,
            'verification': r.verification_type or '—',
        })

    return jsonify(history), 200


# ════════════════════════════════════════════════════════════════
#  6. AGGREGATE ANALYTICS
# ════════════════════════════════════════════════════════════════
@mentor_bp.route('/analytics', methods=['GET'])
def get_mentor_analytics():
    """Returns aggregate analytics across all mentees."""
    mentor_id = _resolve_mentor_id()
    mentee_ids = _get_mentee_ids(mentor_id)

    if not mentee_ids:
        return jsonify({
            'total_mentees': 0,
            'avg_attendance': 0,
            'at_risk_count': 0,
            'satisfactory_count': 0,
            'mentee_rates': [],
        }), 200

    mentee_rates = []
    at_risk = 0
    satisfactory = 0

    for sid in mentee_ids:
        student = User.query.filter_by(user_id=sid).first()
        total = AttendanceRecord.query.filter_by(student_id=sid).count()
        present = AttendanceRecord.query.filter_by(student_id=sid, status='Present').count()
        pct = round((present / total) * 100, 1) if total > 0 else 0.0

        if pct < 75:
            at_risk += 1
        else:
            satisfactory += 1

        mentee_rates.append({
            'student_id': sid,
            'name': student.name if student else 'Unknown',
            'pct': pct,
            'status': 'At Risk' if pct < 75 else 'Satisfactory',
        })

    mentee_rates.sort(key=lambda x: x['pct'])
    avg = round(sum(m['pct'] for m in mentee_rates) / len(mentee_rates), 1) if mentee_rates else 0

    return jsonify({
        'total_mentees': len(mentee_ids),
        'avg_attendance': avg,
        'at_risk_count': at_risk,
        'satisfactory_count': satisfactory,
        'mentee_rates': mentee_rates,
    }), 200


# ════════════════════════════════════════════════════════════════
#  7. AT-RISK ALERTS
# ════════════════════════════════════════════════════════════════
@mentor_bp.route('/alerts', methods=['GET'])
def get_mentor_alerts():
    """Returns mentees below 75% attendance threshold."""
    mentor_id = _resolve_mentor_id()
    mentee_ids = _get_mentee_ids(mentor_id)
    threshold = float(request.args.get('threshold', 75.0))

    alerts = []
    for sid in mentee_ids:
        student = User.query.filter_by(user_id=sid).first()
        if not student:
            continue

        total = AttendanceRecord.query.filter_by(student_id=sid).count()
        present = AttendanceRecord.query.filter_by(student_id=sid, status='Present').count()
        pct = round((present / total) * 100, 1) if total > 0 else 0.0

        if pct < threshold:
            # Find the worst subject
            courses = Course.query.all()
            worst_subj = '—'
            worst_pct = 100.0
            for c in courses:
                enrolled = Enrollment.query.filter_by(student_id=sid, course_id=c.course_id).first()
                if not enrolled:
                    continue
                cr = AttendanceRecord.query.filter_by(student_id=sid, course_id=c.course_id).count()
                cp = AttendanceRecord.query.filter_by(student_id=sid, course_id=c.course_id, status='Present').count()
                cpct = round((cp / cr) * 100, 1) if cr > 0 else 0.0
                if cpct < worst_pct:
                    worst_pct = cpct
                    worst_subj = c.course_name

            alerts.append({
                'student_id': student.user_id,
                'name': student.name,
                'enrollment_no': student.enrollment_no or student.user_id,
                'email': student.email,
                'overall_pct': pct,
                'deficit': round(threshold - pct, 1),
                'worst_subject': worst_subj,
                'worst_subject_pct': worst_pct,
                'severity': 'Critical' if pct < 60 else 'Warning',
            })

    alerts.sort(key=lambda a: a['overall_pct'])

    return jsonify({
        'threshold': threshold,
        'total_alerts': len(alerts),
        'alerts': alerts,
    }), 200


# ════════════════════════════════════════════════════════════════
#  8. MENTOR NOTES — Update notes for a mentee
# ════════════════════════════════════════════════════════════════
@mentor_bp.route('/mentee/<student_id>/notes', methods=['POST'])
def update_mentee_notes(student_id):
    """Update mentor's private notes for a mentee."""
    mentor_id = _resolve_mentor_id()
    mentee_ids = _get_mentee_ids(mentor_id)

    if student_id not in mentee_ids:
        return jsonify({'error': 'Access denied.'}), 403

    data = request.get_json() or {}
    notes = data.get('notes', '')

    assignment = MentorAssignment.query.filter_by(
        mentor_id=mentor_id, student_id=student_id
    ).first()

    if not assignment:
        return jsonify({'error': 'Assignment not found'}), 404

    assignment.notes = notes
    db.session.commit()

    return jsonify({
        'message': 'Notes updated successfully',
        'notes': notes,
    }), 200

File: frontend/js/mentor.js
/**
 * mentor.js — Mentor Dashboard Controller (SPA)
 * Handles mentee overview, individual student detail, attendance analytics,
 * and at-risk alerts. Access is scoped to assigned students only.
 */

import {
  getMentorProfile,
  getMentees,
  getMenteeDetail,
  getMenteeAttendance,
  getMenteeHistory,
  getMentorAnalytics,
  getMentorAlerts,
  updateMenteeNotes,
} from './api.js';

/* ── State ── */
const state = {
  mentorId: 'F001',
  currentSection: 'overview',
  selectedMentee: null,
};

const $ = id => document.getElementById(id);

/* ================================================================
   SPA NAVIGATION
   ================================================================ */
function initNavigation() {
  document.querySelectorAll('.nav-item[data-section]').forEach(item => {
    item.addEventListener('click', e => {
      e.preventDefault();
      switchSection(item.dataset.section);
    });
  });

  $('back-to-mentees')?.addEventListener('click', e => {
    e.preventDefault();
    switchSection('mentees');
  });
}

function switchSection(section) {
  state.currentSection = section;

  document.querySelectorAll('.nav-item[data-section]').forEach(n => n.classList.remove('active'));
  const activeNav = document.querySelector(`.nav-item[data-section="${section}"]`);
  if (activeNav) activeNav.classList.add('active');

  document.querySelectorAll('.dashboard-section').forEach(s => {
    s.style.display = 'none';
    s.classList.remove('active-section');
  });
  const el = $(`section-${section}`);
  if (el) {
    el.style.display = 'block';
    void el.offsetWidth;
    el.classList.add('active-section');
  }

  const titles = {
    overview: 'Mentor Dashboard', mentees: 'My Mentees',
    detail: 'Student Detail', alerts: 'At-Risk Alerts',
  };
  $('topbar-title') && ($('topbar-title').textContent = titles[section] || 'Mentor Dashboard');

  // Show/hide detail nav item
  const detailNav = $('nav-detail');
  if (detailNav) detailNav.style.display = section === 'detail' ? 'flex' : 'none';

  loadSectionData(section);

  // Close mobile sidebar
  $('sidebar')?.classList.remove('open');
  $('sidebar-overlay')?.classList.remove('active');
}

async function loadSectionData(section) {
  switch (section) {
    case 'overview': return loadOverview();
    case 'mentees':  return loadMenteesList();
    case 'detail':   return loadMenteeDetail(state.selectedMentee);
    case 'alerts':   return loadAlerts();
  }
}

/* ================================================================
   SECTION: Overview
   ================================================================ */
async function loadOverview() {
  try {
    const analytics = await getMentorAnalytics(state.mentorId);
    $('ov-total-mentees')  && ($('ov-total-mentees').textContent = analytics.total_mentees || 0);
    $('ov-avg-attendance') && ($('ov-avg-attendance').textContent = `${analytics.avg_attendance || 0}%`);
    $('ov-satisfactory')   && ($('ov-satisfactory').textContent = analytics.satisfactory_count || 0);
    $('ov-at-risk')        && ($('ov-at-risk').textContent = analytics.at_risk_count || 0);
    $('alert-count-badge') && ($('alert-count-badge').textContent = analytics.at_risk_count || 0);

    // Bar chart — mentee rates
    if (analytics.mentee_rates?.length) {
      renderMenteeBarChart(analytics.mentee_rates);
      renderMenteeDonutChart(analytics.satisfactory_count || 0, analytics.at_risk_count || 0);
    }
  } catch (e) {
    console.error('[mentor] overview error:', e);
  }
}

function renderMenteeBarChart(rates) {
  const canvas = $('mentee-bar-chart');
  if (!canvas) return;
  const ctx = canvas.getContext('2d');
  if (canvas._chart) canvas._chart.destroy();

  const labels = rates.map(r => r.name.split(' ')[0]);
  const data = rates.map(r => r.pct);
  const colors = data.map(v => v >= 75 ? '#059669' : v >= 60 ? '#D97706' : '#E11D48');

  canvas._chart = new Chart(ctx, {
    type: 'bar',
    data: { labels, datasets: [{ label: 'Attendance %', data, backgroundColor: colors, borderRadius: 6, barThickness: 24 }] },
    options: {
      responsive: true, maintainAspectRatio: false,
      indexAxis: 'y',
      scales: {
        x: { beginAtZero: true, max: 100, grid: { color: '#F1F5F9' }, ticks: { font: { size: 11 } } },
        y: { grid: { display: false }, ticks: { font: { size: 11, weight: 600 } } }
      },
      plugins: { legend: { display: false } }
    }
  });
}

function renderMenteeDonutChart(satisfactory, atRisk) {
  const canvas = $('mentee-donut-chart');
  if (!canvas) return;
  const ctx = canvas.getContext('2d');
  if (canvas._chart) canvas._chart.destroy();

  canvas._chart = new Chart(ctx, {
    type: 'doughnut',
    data: {
      labels: ['Satisfactory', 'At Risk'],
      datasets: [{
        data: [satisfactory, atRisk],
        backgroundColor: ['#059669', '#E11D48'],
        borderWidth: 0, spacing: 2, borderRadius: 4,
      }]
    },
    options: {
      responsive: true, maintainAspectRatio: false,
      cutout: '65%',
      plugins: {
        legend: { position: 'bottom', labels: { font: { size: 12, weight: 600 }, padding: 16, usePointStyle: true, pointStyleWidth: 10 } }
      }
    }
  });
}

/* ================================================================
   SECTION: Mentees List
   ================================================================ */
async function loadMenteesList(search = '') {
  const grid = $('mentees-grid');
  if (!grid) return;
  grid.innerHTML = '<div class="loading-state"><div class="skeleton skeleton-row"></div><div class="skeleton skeleton-row w-60"></div></div>';

  const data = await getMentees(state.mentorId, search);

  if (!data.mentees.length) {
    grid.innerHTML = '<div class="empty-state" style="padding:40px"><div class="empty-state-title">No mentees found</div><div class="empty-state-desc">No students are assigned to you.</div></div>';
    return;
  }

  grid.innerHTML = data.mentees.map(m => {
    const statusClass = m.status === 'Satisfactory' ? 'satisfactory' : 'at-risk';
    return `
      <div class="mentee-card ${statusClass}" data-student-id="${m.student_id}" onclick="window._viewMentee('${m.student_id}')">
        <div class="mentee-card-header">
          <div class="mentee-card-avatar">${m.initials || m.name.split(' ').map(n => n[0]).join('').slice(0,2).toUpperCase()}</div>
          <div class="mentee-card-info">
            <div class="mentee-card-name">${m.name}</div>
            <div class="mentee-card-enrollment">${m.enrollment_no} · Sem ${m.semester} · Sec ${m.section}</div>
          </div>
          <span class="badge ${m.status === 'Satisfactory' ? 'badge-present' : 'badge-absent'}">${m.attendance_pct}%</span>
        </div>
        <div class="mentee-card-stats">
          <div class="mentee-card-stat">
            <div class="mentee-card-stat-value">${m.present}</div>
            <div class="mentee-card-stat-label">Present</div>
          </div>
          <div class="mentee-card-stat">
            <div class="mentee-card-stat-value">${m.absent}</div>
            <div class="mentee-card-stat-label">Absent</div>
          </div>
          <div class="mentee-card-stat">
            <div class="mentee-card-stat-value">${m.enrolled_courses}</div>
            <div class="mentee-card-stat-label">Courses</div>
          </div>
        </div>
      </div>`;
  }).join('');
}

// Global handler
window._viewMentee = (studentId) => {
  state.selectedMentee = studentId;
  switchSection('detail');
};

let menteeSearchTimeout = null;
function initMenteeSearch() {
  const el = $('mentee-search');
  if (!el) return;
  el.addEventListener('input', () => {
    clearTimeout(menteeSearchTimeout);
    menteeSearchTimeout = setTimeout(() => loadMenteesList(el.value), 300);
  });
}

/* ================================================================
   SECTION: Student Detail
   ================================================================ */
async function loadMenteeDetail(studentId) {
  const container = $('mentee-detail-content');
  if (!container || !studentId) {
    if (container) container.innerHTML = '<div class="empty-state"><div class="empty-state-title">Select a mentee</div></div>';
    return;
  }

  container.innerHTML = '<div class="loading-state"><div class="skeleton skeleton-row"></div><div class="skeleton skeleton-row w-60"></div><div class="skeleton skeleton-row"></div></div>';

  const [detailData, attendanceData, historyData] = await Promise.all([
    getMenteeDetail(studentId, state.mentorId),
    getMenteeAttendance(studentId, state.mentorId),
    getMenteeHistory(studentId, state.mentorId, 15),
  ]);

  if (!detailData || !detailData.profile) {
    container.innerHTML = '<div class="error-state" style="margin:20px"><span>Access denied or student not found.</span></div>';
    return;
  }

  const p = detailData.profile;
  const a = attendanceData;
  const notes = detailData.assignment?.notes || '';

  $('detail-title') && ($('detail-title').textContent = p.name);

  const overallStatus = a.overall_pct >= 75 ? 'Satisfactory' : 'At Risk';
  const overallBadge = a.overall_pct >= 75 ? 'badge-present' : 'badge-absent';

  container.innerHTML = `
    <!-- Profile Panel -->
    <div class="mentee-detail-panel">
      <div class="mentee-detail-header">
        <div class="mentee-detail-avatar">${p.initials || '?'}</div>
        <div class="mentee-detail-info">
          <h2>${p.name}</h2>
          <div class="mentee-detail-meta">${p.enrollment_no} · ${p.program} · Semester ${p.semester} · Section ${p.section}</div>
          <div class="mentee-detail-meta">${p.email}</div>
        </div>
        <span class="badge ${overallBadge}" style="font-size:14px;padding:6px 14px">${a.overall_pct}% Overall</span>
      </div>

      <!-- Quick Stats Row -->
      <div class="stats-grid" style="margin-bottom:0">
        <div class="stat-card primary" style="padding:14px 18px">
          <div class="stat-value" style="font-size:22px">${a.overall_pct}%</div>
          <div class="stat-label">Overall</div>
        </div>
        <div class="stat-card success" style="padding:14px 18px">
          <div class="stat-value" style="font-size:22px">${a.total_present || 0}</div>
          <div class="stat-label">Present</div>
        </div>
        <div class="stat-card danger" style="padding:14px 18px">
          <div class="stat-value" style="font-size:22px">${a.total_classes - (a.total_present || 0)}</div>
          <div class="stat-label">Absent</div>
        </div>
        <div class="stat-card warning" style="padding:14px 18px">
          <div class="stat-value" style="font-size:22px">${a.subjects?.length || 0}</div>
          <div class="stat-label">Subjects</div>
        </div>
      </div>
    </div>

    <!-- Subject-wise Attendance -->
    <h2 style="font-size:16px;font-weight:700;margin-bottom:14px;color:var(--color-text-primary)">Subject-wise Attendance</h2>
    <div class="mentee-subjects-grid" style="margin-bottom:24px">
      ${a.subjects?.map(s => `
        <div class="stat-card ${s.pct >= 75 ? 'success' : 'danger'}" style="padding:16px">
          <div style="font-size:11px;font-weight:700;color:var(--color-primary-accent);letter-spacing:0.5px">${s.code}</div>
          <div style="font-size:14px;font-weight:700;margin:4px 0">${s.name}</div>
          <div style="font-size:12px;color:var(--color-text-muted)">${s.faculty || '—'}</div>
          <div style="display:flex;justify-content:space-between;margin-top:10px;padding-top:10px;border-top:1px solid var(--color-border-light)">
            <span style="font-size:12px;color:var(--color-text-muted)">Present: <strong>${s.present}</strong>/${s.total}</span>
            <span class="badge ${s.pct >= 75 ? 'badge-present' : 'badge-absent'}">${s.pct}%</span>
          </div>
          <div class="progress-track" style="margin-top:8px"><div class="progress-fill ${s.pct >= 75 ? 'high' : s.pct >= 60 ? 'medium' : 'low'}" style="width:${s.pct}%"></div></div>
        </div>
      `).join('') || '<p style="color:var(--color-text-muted)">No subject data available.</p>'}
    </div>

    <!-- Recent History -->
    <h2 style="font-size:16px;font-weight:700;margin-bottom:14px;color:var(--color-text-primary)">Recent Attendance History</h2>
    <div style="background:var(--color-surface);border:1px solid var(--color-border);border-radius:var(--radius-lg);overflow:hidden;margin-bottom:24px">
      <div class="table-container">
        <table>
          <thead><tr><th>Date</th><th>Subject</th><th>Code</th><th>Status</th><th>Time</th></tr></thead>
          <tbody>
            ${historyData.length ? historyData.map(h => `
              <tr>
                <td class="td-mono">${h.date}</td>
                <td class="td-primary">${h.subject}</td>
                <td class="td-mono">${h.code}</td>
                <td><span class="badge ${h.status === 'Present' ? 'badge-present' : h.status === 'Flagged' ? 'badge-flagged' : 'badge-absent'}">${h.status}</span></td>
                <td style="font-size:12px;color:var(--color-text-muted)">${h.time}</td>
              </tr>
            `).join('') : '<tr><td colspan="5" style="text-align:center;padding:20px;color:var(--color-text-muted)">No history available</td></tr>'}
          </tbody>
        </table>
      </div>
    </div>

    <!-- Mentor Notes -->
    <h2 style="font-size:16px;font-weight:700;margin-bottom:14px;color:var(--color-text-primary)">Mentor Notes</h2>
    <div style="background:var(--color-surface);border:1px solid var(--color-border);border-radius:var(--radius-lg);padding:20px">
      <textarea class="notes-textarea" id="mentee-notes" placeholder="Add private notes about this student...">${notes}</textarea>
      <div style="display:flex;justify-content:flex-end;margin-top:10px">
        <button class="btn btn-primary" id="save-notes-btn">Save Notes</button>
      </div>
    </div>
  `;

  // Wire save notes
  $('save-notes-btn')?.addEventListener('click', async () => {
    const notesVal = $('mentee-notes')?.value || '';
    const result = await updateMenteeNotes(studentId, notesVal, state.mentorId);
    if (!result.error) showToast('Notes saved successfully');
    else showToast('Failed to save notes');
  });
}

/* ================================================================
   SECTION: At-Risk Alerts
   ================================================================ */
async function loadAlerts() {
  const list = $('alerts-list');
  if (!list) return;
  list.innerHTML = '<div class="loading-state"><div class="skeleton skeleton-row"></div></div>';

  const data = await getMentorAlerts(state.mentorId);
  $('alert-total-badge')  && ($('alert-total-badge').textContent = `${data.total_alerts} students`);
  $('alert-count-badge')  && ($('alert-count-badge').textContent = data.total_alerts);

  if (!data.alerts.length) {
    list.innerHTML = '<div class="empty-state" style="padding:40px"><div class="empty-state-icon" style="color:var(--color-success)"><svg width="40" height="40" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2"><polyline points="20 6 9 17 4 12"/></svg></div><div class="empty-state-title" style="color:var(--color-success)">All Clear!</div><div class="empty-state-desc">All your mentees are above the 75% attendance threshold.</div></div>';
    return;
  }

  list.innerHTML = data.alerts.map(a => `
    <div class="alert-card ${a.severity.toLowerCase()}" style="cursor:pointer" onclick="window._viewMentee('${a.student_id}')">
      <div class="alert-card-icon ${a.severity.toLowerCase()}">
        <svg width="20" height="20" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2"><path d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"/><line x1="12" y1="9" x2="12" y2="13"/><line x1="12" y1="17" x2="12.01" y2="17"/></svg>
      </div>
      <div class="alert-card-content">
        <div class="alert-card-name">${a.name} <span style="font-weight:400;color:var(--color-text-muted)">(${a.enrollment_no})</span></div>
        <div class="alert-card-detail">Worst: ${a.worst_subject} at ${a.worst_subject_pct}% · Deficit: -${a.deficit}%</div>
      </div>
      <div class="alert-card-pct ${a.severity.toLowerCase()}">${a.overall_pct}%</div>
    </div>
  `).join('');
}

/* ================================================================
   UI HELPERS
   ================================================================ */
function showToast(msg) {
  const container = $('toast-container');
  if (!container) return;
  const toast = document.createElement('div');
  toast.className = 'toast';
  toast.innerHTML = `<svg width="14" height="14" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2"><polyline points="20 6 9 17 4 12"/></svg> <span>${msg}</span>`;
  container.appendChild(toast);
  setTimeout(() => toast.remove(), 3000);
}

function initSidebar() {
  const toggle = $('sidebar-toggle'), sidebar = $('sidebar'), overlay = $('sidebar-overlay');
  if (toggle && sidebar) {
    toggle.addEventListener('click', () => { sidebar.classList.toggle('open'); overlay?.classList.toggle('active'); });
  }
  overlay?.addEventListener('click', () => { sidebar?.classList.remove('open'); overlay.classList.remove('active'); });
  $('sidebar-logout-btn')?.addEventListener('click', () => { showToast('Logging out...'); setTimeout(() => window.location.href = 'index.html', 800); });
}

/* ════════ Boot ════════ */
async function init() {
  initSidebar();
  initNavigation();
  initMenteeSearch();

  $('header-date') && ($('header-date').textContent = new Date().toLocaleDateString('en-IN', { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' }));

  try {
    const profile = await getMentorProfile(state.mentorId);
    $('mentor-name')   && ($('mentor-name').textContent = profile.name);
    $('mentor-dept')   && ($('mentor-dept').textContent = profile.department);
    $('mentor-avatar') && ($('mentor-avatar').textContent = profile.initials);
  } catch (e) {}

  switchSection('overview');
}

document.addEventListener('DOMContentLoaded', init);
